alkira/resource_alkira_connector_azure_expressroute_helper.go

```diff
@@ -74,3 +184,80 @@ func expandAzureExpressRouteSegments(seg []interface{}, m interface{}) ([]alkira
 
 	return segments, nil
 }
+func flattenInstance(instance alkira.ConnectorAzureExpressRouteInstance, m interface{}) map[string]interface{} {
+	result := map[string]interface{}{
+		"credential_id":             instance.CredentialId,
+		"expressroute_circuit_id":   instance.ExpressRouteCircuitId,
+		"gateway_mac_address":       instance.GatewayMacAddress,
+		"id":                        instance.Id,
+		"loopback_subnet":           instance.LoopbackSubnet,
+		"name":                      instance.Name,
+		"redundant_router":          instance.RedundantRouter,
+		"virtual_network_interface": instance.Vnis,
+		"ipsec_customer_gateway":    flattenInstanceSegmentOptions(instance.SegmentOptions, m),
+	}
+	return result
+}
+
+// flattenInstanceSegmentOptions flattens the segment options for an instance
+func flattenInstanceSegmentOptions(segmentOptions []alkira.InstanceSegmentOption, m interface{}) []interface{} {
+	if segmentOptions == nil {
+		return nil
+	}
+
+	result := make([]interface{}, len(segmentOptions))
+	for i, segOpt := range segmentOptions {
+		segmentId, err := getSegmentIdByName(segOpt.SegmentName, m)
+		if err != nil {
+			log.Printf("[DEBUG] Unable to find segment ID for name %s: %v", segOpt.SegmentName, err)
```

Comment:
This error message won't help much... No one will see it.. It's better to pop the `Error` up and then fail it, so return a error from the function and then fail the provider.